+nigeLab/@nigelObj/nigelObj.m

```diff
@@ -5055,45 +5067,15 @@ function checkAnimalsForClones(obj,~,~)
          % cycle through each animal, removing animals and adding any
          % associated blocks to the "kept" animal Blocks property
          ii=1;
-         while ~isempty(comparisons_mat)
-            % Current row contains all comparisons to other Animals in
-            % obj.Children
-            animalIsSame = comparisons_mat(1,:);
-            comparisons_mat(1,:) = []; % ensure this row is dropped
-            
-            % ii indexes current "good" Animal
-            a = A(ii);
-            ii = ii + 1; % ensure it is incremented
-            
-            % If no redundancies, then continue.
-            if ~any(animalIsSame)
-               continue;
-            end
-            
-            % To prevent weird case where you have a 1x0 array
-            if isempty(animalIsSame)
-               continue;
-            end
-            
-            % Add child blocks from removed animals to this animal to
-            % ensure they aren't accidentally discarded
-            aidx = find(animalIsSame);
-            B = [A(aidx).Children]; %#ok<*FNDSB>
-            addChild(a,B);
-            
-            % Now, remove redundant animals from array and also remove them
-            % from the comparisons matrix since we don't need to redo them
-            mask = find(idx);
-            obj.Children(mask(animalIsSame)) = []; % Remove from property
-            A(animalIsSame) = []; % Remove them from consideration in the array
-            idx(animalIsSame) = []; % Remove corresponding indexes
-            
-            % Lastly, update the comparisons matrices
-            iRow = animalIsSame(2:end); % To account for previously-removed row of comparisons
-            comparisons_mat(iRow,:) = [];
-            % Columns are not removed, since the original animal is kept in
-            % the array and we should account for its index.
-            comparisons_mat(:,animalIsSame) = [];
+         while any(rmvec)             
+             idxToRemove = find(rmvec,1);
+             idxToKeep = find(correspondanceVec,1);
+             a = A(idxToKeep);
+             B = [A(idxToRemove).Children]; %#ok<*FNDSB>
+             addChild(a,B);
+             obj.Children(idxToRemove) = []; % Remove from property
+             rmvec(idxToRemove) = [];
+             correspondanceVec(idxToKeep) = [];
```

Comment:
I remember having issues with this when I initially wrote it, then re-wrote it to try and simplify it, then eventually reverted back to the older "less-elegant" code. Have you tested this out on different use-cases? One issue could have been during `reload` if we were reloading transient Animal objects as `null`, then this would automatically remove them.